### ISSUE TITLE
DecoderTransformer not implemented as paper at all

### ISSUE THREAD

**Issue by mvccn** [AIStream-Peelout/flow-forecast#470]
did I miss something? The decodertransformer which claims to implement the paper(Enhancing the Locality and Breaking the Memory Bottleneck of Transformer on Time Series Forecasting) in the document is not even close to what the paper proposed. There are no key component like conv1d layers for locality and logsparse. If we didn't implement that paper, then really shouldn't list it in the document.

**Comment by isaacmg** [comment]
>  in the document is not even close to what the paper proposed. There are no key component like conv1d layers for locality and logsparse.

[Yes there is](https://github.com/AIStream-Peelout/flow-forecast/blob/d37ed04c27cc14b7dc5a66af001255b1a1bb99c7/flood_forecast/transformer_xl/transformer_bottleneck.py#L56) . Also the code for the DecoderTransformer was supplied to me directly by the papers original authors. The one thing that arguably is lacking is end-to-end training with the series_id as FF data loaders lack the ability to supply series id #307  however #342 will address this.

**Comment by mvccn** [comment]
sorry, my bad. I see them in class attention. Thanks a lot for the quick reply. Another note, do you know why the authors only "conv, k" queries and keys(K,Q), but use "conv, 1" for values(V)? The paper doesn't explain this, I am wondering if "conv,k" V would also work.

**Comment by isaacmg** [comment]
I'm not sure why they chose not to modify the value vectors the same way they do with the query and key vectors. But this seems consistent with there statement that:

> We propose convolutional self-attention by employing causal convolutions to produce queries and
keys in the self-attention layer. Query-key matching aware of local context, e.g. shapes, can help
the model achieve lower training loss and further improve its forecasting accuracy.

Where they make discuss using convolution for the queries and keys in the self-attention but imply they don't change the value vectors..

**Comment by mvccn** [comment]
yes, I saw that too. closing this, thanks for your help.
